js/render/painter/use_program.js

```diff
@@ -119,27 +120,19 @@ module.exports.useProgram = function (nextProgramName, posMatrix, exMatrix) {
     if (previousProgram !== nextProgram) {
         gl.useProgram(nextProgram.program);
 
-        // Disable all attributes from the existing program that aren't used in
-        // the new program. Note: attribute indices are *not* program specific!
-        var nextAttributes = util.objectValues(nextProgram.attributes);
-        var previousAttributes = util.objectValues((previousProgram && previousProgram.attributes) || {});
-
-        for (var i = 0; i < previousAttributes.length; i++) {
-            if (nextAttributes.indexOf(previousAttributes[i]) < 0) {
-
-                // WebGL breaks if you disable attribute 0.
-                // http://stackoverflow.com/questions/20305231
-                assert(previousAttributes[i] !== 0);
+        var numNextAttributes = nextProgram.numAttributes;
+        var numPrevAttributes = previousProgram ? previousProgram.numAttributes : 0;
+        var i;
 
-                gl.disableVertexAttribArray(previousAttributes[i]);
-            }
+        // Disable all attributes from the previous program that aren't used in
+        // the new program. Note: attribute indices are *not* program specific!
+        // WebGL breaks if you disable attribute 0. http://stackoverflow.com/questions/20305231
+        for (i = Math.max(1, numNextAttributes); i < numPrevAttributes; i++) {
+            gl.disableVertexAttribArray(i);
```

Comment:
it might be good to re-add this assertion https://github.com/mapbox/mapbox-gl-js/blob/a6c406cd45411151dc2ec1151835ff2e7060b57f/js/render/painter/use_program.js#L130